ide/extbrowser/src/org/netbeans/modules/extbrowser/NbDdeBrowserImpl.java

```diff
@@ -127,6 +133,59 @@ public NbDdeBrowserImpl (ExtWebBrowser extBrowserFactory) {
      */
     public static native String getDefaultOpenCommand() throws NbBrowserException;
     
+        /**
+     * Get the default browser name using Java JNA library
+     * @return String
+     */
+    private static String getDefaultWindowsBrowser() {
+        String userChoice = Advapi32Util
+                .registryGetStringValue(
+                        WinReg.HKEY_CURRENT_USER,
+                        "SOFTWARE\\Microsoft\\Windows\\Shell\\Associations\\UrlAssociations\\https\\UserChoice",
+                        "ProgId"
+                )
+                .toUpperCase(Locale.ROOT);
```

Comment:
this will throw a NPE if `registryGetStringValue(...)` returns null

change to:
```java
        String userChoice = Advapi32Util
                .registryGetStringValue(
                        WinReg.HKEY_CURRENT_USER,
                        "SOFTWARE\\Microsoft\\Windows\\Shell\\Associations\\UrlAssociations\\https\\UserChoice",
                        "ProgId"
                );

        if (userChoice == null || userChoice.trim().isEmpty()) {
            return ExtWebBrowser.IEXPLORE;
        } else {
            userChoice = userChoice.toUpperCase(Locale.ROOT);
        }
```